rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -205,12 +206,21 @@ class ParticipantListener : public eprosima::fastdds::dds::DomainParticipantList
         rmw_qos_profile_t qos_profile = rmw_qos_profile_unknown;
         rtps_qos_to_rmw_qos(proxyData.m_qos, &qos_profile);
 
+        const auto & userDataValue = proxyData.m_qos.m_userData.getValue();
+        rosidl_type_hash_t type_hash;
+        if (RMW_RET_OK != rmw_dds_common::parse_type_hash_from_user_data(
+            userDataValue.data(), userDataValue.size(), type_hash))
+        {
+          type_hash = rosidl_get_zero_initialized_type_hash();
```

Comment:
As I wrote in https://github.com/ros2/rmw_fastrtps/pull/671#discussion_r1138753995 , I think we should probably print a WARNING here.